meson.build

```
@@ -0,0 +1,153 @@
+  'src/src/commands/ush_cmd_pwd.c',
+  'src/src/commands/ush_cmd_cat.c',
+  'src/src/commands/ush_cmd_xxd.c',
+  'src/src/commands/ush_cmd_echo.c'
+]
+
+project_dependencies = [
+]
+
+include_directories = [
+]
+
+public_headers = include_directories('src')
+
+build_args = [
+  '-Werror',
+  '-Wall',
+  '-Wextra',
+  '-pedantic',
+]
+
+link_args = [
+]
+
+settings_map = {
+  'filename-align-space':        '-DUSH_CONFIG_FILENAME_ALIGN_SPACE',
+  'cmd-xxd-columns':             '-DUSH_CONFIG_CMD_XXD_COLUMNS',
+}
+
+feature_map = {
+  'enable-command-cat':          '-DUSH_CONFIG_ENABLE_COMMAND_CAT',
+  'enable-command-cd':           '-DUSH_CONFIG_ENABLE_COMMAND_CD',
+  'enable-command-echo':         '-DUSH_CONFIG_ENABLE_COMMAND_ECHO',
+  'enable-command-help':         '-DUSH_CONFIG_ENABLE_COMMAND_HELP',
+  'enable-command-pwd':          '-DUSH_CONFIG_ENABLE_COMMAND_PWD',
+  'enable-command-xxd':          '-DUSH_CONFIG_ENABLE_COMMAND_XXD',
+  'enable-command-ls':           '-DUSH_CONFIG_ENABLE_COMMAND_LS',
+
+  'enable-feature-commands':     '-DUSH_CONFIG_ENABLE_FEATURE_COMMANDS',
+  'enable-feature-autocomplete': '-DUSH_CONFIG_ENABLE_FEATURE_AUTOCOMPLETE',
+  'enable-feature-shell-styles': '-DUSH_CONFIG_ENABLE_FEATURE_SHELL_STYLES',
+}
+
+translation_map = {
+  'translation-ok':                        '-DUSH_CONFIG_TRANSLATION_OK',
+  'translation-error':                     '-DUSH_CONFIG_TRANSLATION_ERROR',
+  'translation-directory-not-found':       '-DUSH_CONFIG_TRANSLATION_DIRECTORY_NOT_FOUND',
+  'translation-nested-directories-exist':  '-DUSH_CONFIG_TRANSLATION_NESTED_DIRECTORIES_EXIST',
+  'translation-cannot-find-parent-node':   '-DUSH_CONFIG_TRANSLATION_CANNOT_FIND_PARENT_NODE',
+  'translation-directory-already-mounted': '-DUSH_CONFIG_TRANSLATION_DIRECTORY_ALREADY_MOUNTED',
+  'translation-syntax-error':              '-DUSH_CONFIG_TRANSLATION_SYNTAX_ERROR',
+  'translation-wrong-arguments':           '-DUSH_CONFIG_TRANSLATION_WRONG_ARGUMENTS',
+  'translation-file-not-executable':       '-DUSH_CONFIG_TRANSLATION_FILE_NOT_EXECUTABLE',
+  'translation-file-not-writable':         '-DUSH_CONFIG_TRANSLATION_FILE_NOT_WRITABLE',
+  'translation-file-not-readable':         '-DUSH_CONFIG_TRANSLATION_FILE_NOT_READABLE',
+  'translation-no-help-available':         '-DUSH_CONFIG_TRANSLATION_NO_HELP_AVAILABLE',
+  'translation-file-not-found':            '-DUSH_CONFIG_TRANSLATION_FILE_NOT_FOUND',
+  'translation-read-only-file':            '-DUSH_CONFIG_TRANSLATION_READ_ONLY_FILE',
+}
+
+foreach option_key, option_value : settings_map
+  add_project_arguments(option_value + '=' + get_option(option_key).to_string(), language: 'c')
+  summary(option_key, get_option(option_key), section: 'Enabled Settingss')
+endforeach
+
+foreach option_key, option_value : feature_map
+  if get_option(option_key)
+    add_project_arguments(option_value, language: 'c')
+  endif
+  summary(option_key, get_option(option_key), section: 'Enabled Features')
```

```suggestion
  summary(option_key, get_option(option_key), section: 'Enabled Features', bool_yn: true)
```

May look a bit nicer IMO.